version: '3.8'
services:

  app:
    build:
      context: .
      dockerfile: Dockerfile.pytest.app
    ports:
      - "8000:8000"
    container_name: app
    environment:
      TEST_DATABASE_URL: 'sqlite+aiosqlite:///:memory:'
      API_KEY_SECRET: laisjdf;asdifj[fi9wejr'wekf]
      ENCRYPTION_KEY_SECRET: alsk;fj2 i3jeqealfjansdflkmadf
    volumes:
      - ./pytest/reports:/usr/src/app/pytest/reports
      - ./pytest/data:/usr/src/app/pytest/data
    networks:
      - app-network

  nginx:
    image: nginx:alpine
    volumes:
      - ./pytest/reports/coverage/htmlcov:/usr/share/nginx/html:ro
    ports:
      - "8080:80"
    depends_on:
      - app
    networks:
      - app-network

  allure:
    image: frankescobar/allure-docker-service
    ports:
      - "4040:4040"
    volumes:
      - ./pytest/reports/allure-results:/app/allure-results
      - ./pytest/reports/allure-report:/app/allure-report
    environment:
      CHECK_RESULTS_EVERY_SECONDS: 10
      KEEP_HISTORY: 1
    depends_on:
      - app
    networks:
      - app-network

networks:
  app-network:
    driver: bridge
